agent/command/s3_copy.go

```diff
@@ -223,20 +232,6 @@ func (c *s3copy) copyWithRetry(ctx context.Context,
 			continue
 		}
 
-		if len(s3CopyFile.BuildVariants) > 0 && !utility.StringSliceContains(
-			s3CopyFile.BuildVariants, conf.BuildVariant.Name) {
-			continue
-		}
-
-		if ctx.Err() != nil {
-			return errors.New("s3copy operation received was canceled")
-		}
-
-		logger.Execution().Infof("Making API push copy call to "+
-			"transfer %v/%v => %v/%v", s3CopyFile.Source.Bucket,
-			s3CopyFile.Source.Path, s3CopyFile.Destination.Bucket,
-			s3CopyFile.Destination.Path)
-
```

Comment:
Should we be updating push status to failed for each of the error return statements after this? (or maybe handling in a defer?) It looks like we would leave it as "pushing" if we exited anytime before the retry loop

Reply:
good point, will move.